jobqueue/scheduler/openstack.go

```diff
@@ -827,12 +837,8 @@ func (s *opst) reqForSpawn(req *Requirements) *Requirements {
 
 // spawn creates a new instance in OpenStack. Errors are not returned but are
 // logged, and problematic servers are terminated.
-func (s *opst) spawn(req *Requirements, flavor *cloud.Flavor, requestedOS string, requestedScript []byte, requestedConfigFiles string, needsSharedDisk bool, cmd string, call string) {
-	// since we can have many simultaneous calls to this method running at once,
-	// we make a new logger with a unique "call" context key to keep track of
-	// which spawn call is doing what
-	logger := s.Logger.New("call", call, "flavor", flavor.Name)
```

Comment:
Instead of taking a call value, the caller of spawn should set call value in the ctx.

So it looks like we need new clog methods to add a unique call value, and a server flavour.